### PR TITLE
git-lfs-transfer: init at 0.1.0-unstable-2024-10-07

### PR DESCRIPTION
**Pull request by CHN-beta**
This package is needed to use git lfs over pure ssh when the server does not have HTTPS support.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/gi/git-lfs-transfer/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule {
+  pname = "git-lfs-transfer";
+  version = "unstable-2024-10-07";
```
Comment (loicreynier) [inline pkgs/by-name/gi/git-lfs-transfer/package.nix:8]:
```suggestion
  version = "0.1.0-unstable-2024-10-07";
```

[Version should start with a digit](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning), last version was 0.1.0.

File: pkgs/by-name/gi/git-lfs-transfer/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    owner = "charmbracelet";
+    repo = "git-lfs-transfer";
+    rev = "422d24414fe4b803849b3f6fe7c4d8ab1b40803b";
+    hash = "sha256-YsplPW3i4W1RfkWQI1eGXFXb3JofQwKe+9LbjxeL1cM=";
+  };
+
+  vendorHash = "sha256-1cGlhLdnU6yTqzcB3J1cq3gawncbtdgkb3LFh2ZmXbM=";
+
+  meta = {
+    description = "Server-side implementation of the Git LFS pure-SSH protocol.";
```
Comment (FliegendeWurst) [inline pkgs/by-name/gi/git-lfs-transfer/package.nix:20]:
```suggestion
    description = "Server-side implementation of the Git LFS pure-SSH protocol";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/gi/git-lfs-transfer/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    hash = "sha256-YsplPW3i4W1RfkWQI1eGXFXb3JofQwKe+9LbjxeL1cM=";
+  };
+
+  vendorHash = "sha256-1cGlhLdnU6yTqzcB3J1cq3gawncbtdgkb3LFh2ZmXbM=";
+
+  meta = {
+    description = "Server-side implementation of the Git LFS pure-SSH protocol";
+    mainProgram = "git-lfs-transfer";
+    homepage = "https://github.com/charmbracelet/git-lfs-transfer";
+    license = with lib.licenses; [ mit ];
```
Comment (Aleksanaa) [inline pkgs/by-name/gi/git-lfs-transfer/package.nix:23]:
```suggestion
    license = lib.licenses.mit;
```

File: pkgs/by-name/gi/git-lfs-transfer/package.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "charmbracelet";
+    repo = "git-lfs-transfer";
+    rev = "422d24414fe4b803849b3f6fe7c4d8ab1b40803b";
+    hash = "sha256-YsplPW3i4W1RfkWQI1eGXFXb3JofQwKe+9LbjxeL1cM=";
+  };
+
+  vendorHash = "sha256-1cGlhLdnU6yTqzcB3J1cq3gawncbtdgkb3LFh2ZmXbM=";
+
```
Comment (Aleksanaa) [inline pkgs/by-name/gi/git-lfs-transfer/package.nix:18]: Would be good to add https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook if possible

Comment (FliegendeWurst) [reply pkgs/by-name/gi/git-lfs-transfer/package.nix:18]: Unfortunately the binary has no --version

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**